examples/custom-node/src/primitives/tx.rs

```diff
@@ -174,12 +171,14 @@ impl Compact for CustomTransaction {
     where
         B: alloy_rlp::bytes::BufMut + AsMut<[u8]>,
     {
-        self.inner.to_compact(buf)
+        let start_len = buf.remaining_mut();
+        self.inner.tx().encode(buf);
+        start_len - buf.remaining_mut()
```

Comment:
here we're losing the signature

we should encode this like self.signature().to_compact
self.inner.tx().to_compact